pages/my-account/index.vue

```diff
@@ -111,17 +144,120 @@ export default defineComponent({
         app.router.push({ path: "/" })
         return false
       }
+      return false
     }
 
     const gotoOrderHistory = () => {
       app.router.push({ path: "my-account/order-history" })
     }
 
-    /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
-    const saveAddress = (address) => {
-      // add/save address here
+    const saveAddress = async ({ address, setAsDefault }, typeName) => {
+      const addressInputFormat = {
+        accountId: user.value.id,
+        customerContactInput: { ...address },
+      }
+      if (address.id) {
+        // update scenario
+        addressInputFormat.contactId = address.id
+        address.types.find((t) => t.name === typeName).isPrimary = setAsDefault
+
+        await updateUserAddress(addressInputFormat)
+      } else {
+        // add new scenarion
+        addressInputFormat.customerContactInput.types = [
+          {
+            name: typeName,
+            isPrimary: setAsDefault,
+          },
+        ]
+        addressInputFormat.customerContactInput.accountId = user.value.id
+        await addUserAddress(addressInputFormat)
+      }
+    }
+
+    const saveShippingAddress = async (params) => {
+      await saveAddress(params, "Shipping")
+      await loadUserAddresses(user.value.id)
     }
 
+    const saveBillingAddress = async (params) => {
+      await saveAddress(params, "Billing")
+      await loadUserAddresses(user.value.id)
+    }
+
+    const deleteAddress = async (address) => {
+      if (!address.id) {
+        alert("Sorry no address found !!")
+        return false
+      }
```

Comment:
In which scenario this code block will get execute? preferably use model dialog instead of browser alert.

Reply:
You are correct good catch, removed..